GatherBuddy/AutoGather/AutoGather.cs

```diff
@@ -83,12 +88,33 @@ public void GoHome()
             WentHome = true;
 
             if (!GatherBuddy.Config.AutoGatherConfig.GoHomeWhenIdle)
+            {
+                AutoGatherStatus = AutoGatherState.Idle;
                 return;
+            }
 
             if (Lifestream_IPCSubscriber.IsEnabled && !Lifestream_IPCSubscriber.IsBusy())
+            {
                 Lifestream_IPCSubscriber.ExecuteCommand("auto");
-            else 
+                Task.Run(() =>
+                {
+                    const int maxLoops = 60;
+                    int count = 0;
+                    while (Lifestream_IPCSubscriber.IsBusy() && count < maxLoops)
+                    {
+                        Thread.Sleep(1000);
+                        count++;
+                    }
+                    // Wait until Lifestream successfully reaches it's destination before marking AutoGatherStatus as Idle
+                    // This ensure that Artisan isnt restarted while Lifestream is Busy
+                    AutoGatherStatus = AutoGatherState.Idle;
+                });
```

Comment:
I'm not familiar with the ECommons library. There may be a better alternative for asynchronous task queuing that I'm not aware of.

Reply:
Use TaskManager.Enqueue. It can repeatedly check a boolean expression until it is true, and also has a timeout.

Reply:
TaskManager.Enqueue(() => !Lifestream_IPCSubscriber.IsBusy(), 60000);
TaskManager.Enqueue(() =>{AutoGatherStatus = AutoGatherState.Idle;});

Reply:
Thanks for the suggestion